data-prepper-plugins/otel-trace-raw-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltrace/OTelTraceRawPrepper.java

```diff
@@ -247,6 +249,37 @@ private boolean shouldGarbageCollect() {
         return System.currentTimeMillis() - lastTraceFlushTime >= traceFlushInterval;
     }
 
+    /**
+     * Re-enqueues all spans with time "0" so that all will be available for consumption.
+     */
+    @Override
+    public void prepareForShutdown() {
+        boolean isLockAcquired = prepareForShutdownLock.tryLock();
+
+        if (isLockAcquired) {
+            try {
+                LOG.info("Preparing for shutdown, re-enqueueing {} spans", delayedParentSpanQueue.size());
+                Iterator iterator = delayedParentSpanQueue.iterator();
+                List<DelayedParentSpan> delayedParentSpanList = ImmutableList.copyOf(iterator);
```

Comment:
QUES: is there reason we do not use drainTo here?

Reply:
Yes - drainTo only transfers objects that meet the delay criteria. 

Using an iterator allows us to fetch ALL objects in the queue regardless of the delay values.